src/mscorlib/shared/System/IO/Path.Windows.cs

```diff
@@ -223,5 +223,42 @@ public static ReadOnlySpan<char> GetPathRoot(ReadOnlySpan<char> path)
 
         /// <summary>Gets whether the system is case-sensitive.</summary>
         internal static bool IsCaseSensitive { get { return false; } }
+
+        internal static ReadOnlySpan<char> GetVolumeName(ReadOnlySpan<char> path)
+        {
+            if (!IsPathFullyQualified(path))
+                return default;
+
+            // 3 cases: UNC ("\\server\share"), Device ("\\?\C:\"), or Dos ("C:\")
+            ReadOnlySpan<char> root = GetPathRoot(path);
+            if (!PathInternal.IsDevice(path))
+                return TrimEndingDirectorySeparator(root); // e.g. "C:"
+
+            if (IsUnc(path))
+            {
+                // Cut from "\\?\UNC\Server\Share" to "Server\Share"
+                return TrimEndingDirectorySeparator(root.Slice(8));
+            }
+            else
+            {
+                // Cut from "\\?\C:\" to "C:"
+                return TrimEndingDirectorySeparator(root.Slice(4));
+            }
+        }
+
+        internal static bool EndsInDirectorySeparator(ReadOnlySpan<char> path)
+        {
+            return path.Length > 0 && PathInternal.IsDirectorySeparator(path[path.Length - 1]);
+        }
+
+        internal static ReadOnlySpan<char> TrimEndingDirectorySeparator(ReadOnlySpan<char> path) =>
+            EndsInDirectorySeparator(path) ?
+                path.Slice(0, path.Length - 1) :
+                path;
+
+        internal static bool IsUnc(ReadOnlySpan<char> path) =>
```

Comment:
UNCs are in three formats. `\\Server\Share`, `\\?\UNC\Server\Share`, or `\\.\UNC\Server\Share`.  Anything that begins with `\\` and is _not_ followed by `.` or `?` is a UNC. If it is followed by `.` or `?` it needs to have at least `\UNC\` to be a UNC. Technically a UNC also requires at least two segments (for the server and share) past the prefix to be valid but we don't care about that here.